cpp/oneapi/dal/table/backend/homogen_table_builder_impl.hpp

```diff
@@ -121,11 +147,20 @@ class homogen_table_builder_impl
                    const void* data,
                    std::int64_t row_count,
                    std::int64_t column_count) override {
-        auto& queue = policy.get_queue();
-        ONEDAL_ASSERT(sycl::get_pointer_type(data_.get_data(), queue.get_context()) !=
-                      sycl::usm::alloc::unknown);
         check_copy_data_preconditions(row_count, column_count);
-        detail::memcpy(queue, data_.get_mutable_data(), data, data_.get_size());
+
+        auto& input_q = policy.get_queue();
+        auto this_q = data_.get_queue();
```

Comment:
You need to check if both queues are in the same context, see 
https://github.com/oneapi-src/oneDAL/blob/80e130dae144cdc378acef095839bd0f92c742a9/cpp/oneapi/dal/backend/common.hpp#L282

Reply:
discussed offline